src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -13,10 +13,14 @@ public static DocumentUpdater construct(UpdateClause updateDef) {
     return new DocumentUpdater(updateDef.buildOperations());
   }
 
-  public JsonNode applyUpdates(JsonNode readDocument) {
+  public DocumentUpdaterResponse applyUpdates(JsonNode readDocument) {
     UpdateTargetLocator targetLocator = new UpdateTargetLocator();
     ObjectNode docToUpdate = (ObjectNode) readDocument;
-    updateOperations.forEach(u -> u.updateDocument(docToUpdate, targetLocator));
-    return readDocument;
+    boolean modified = false;
+    for (UpdateOperation updateOperation : updateOperations)
+      modified |= updateOperation.updateDocument(docToUpdate, targetLocator);
```

Comment:
Is this safe? If `modified` is `true`, would this not short-circuit and skip call to `updateOperation.upateDocument(...)` completely?

Reply:
Yes, it runs the method before evaluating applying or condition.
The below code should print the numbers 0, 1 ... 9.
`public class Test{
  public static boolean update(int i){
        System.out.println(i);
        return i%2 ==0;
    }
    public static void main(String args[]){
        boolean modified = false;
        for(int i = 0; i < 10; i++){
            modified |= update(i);
       }
  }
}`